staging/src/k8s.io/client-go/tools/cache/object-names.go

```diff
@@ -0,0 +1,65 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cache
+
+import (
+	"k8s.io/apimachinery/pkg/types"
+)
+
+// ObjectName is a reference to an object of some implicit kind
+type ObjectName struct {
```

Comment:
Do we get value from another type vs just using NamespacedName?

Reply:
Yes, `types.NamespacedName`. It is already the same shape as this struct, and you have conversion function in this file defined. I am not sure why we want 2 structs for the same purpose, with the same fields.

Reply:
Oh, sorry, I misunderstood the question.  We need a type whose `String` function matches the historical behavior of `MetaNamespaceKeyFunc`, to maximize continuity.

Reply:
The comment _is_ in the code, on the `String` method.

I would not try to deprecate a lower level type, I think that would be too disruptive.